blocks/bricks/recurrent/base.py

```diff
@@ -46,6 +46,9 @@ def initial_states(self, batch_size, *args, **kwargs):
             The keyword arguments of the application call.
 
         """
+        if not hasattr(self, 'apply') or not self.apply.states:
+            return
+
```

Comment:
when some subclass call the default `initial_states` function in the `BaseRecurrent` class. This line would check whether it is necessary to return the initial states. If the subclass does not have an `apply` method or its `apply` method does not contain `states`, the `initial_states` would not return anything.
This line would make it to support `recurrent` class with no `apply` function or with no `states`.